frontend/src/components/messages/BasicMessages.vue

```diff
@@ -0,0 +1,178 @@
+  import { mapMutations } from 'vuex';
+  import {CHAT_CURRENT_USERID} from "@/constants";
+  import {formatDateLong} from "@/filters";
+
+  export default {
+    components: {
+      ChatWindow
+    },
+    created() {
+    },
+    data() {
+      return {
+        currentUserId: CHAT_CURRENT_USERID,
+        responsiveBreakpoint: 9999,
+        rooms: [],
+        roomsLoaded: true,
+        loadingRooms: false,
+        roomId: null,
+        messages: [],
+        messagesLoaded: true,
+        messageActions: [],
+        textMessages: {
+          CONVERSATION_STARTED: ""
+        }
+      }
+    },
+    computed: {
+      partnersCount() {
+        return this.$store.getters.partnersCount;
+      },
+      messagesCount() {
+        return this.$store.getters.messagesCount;
+      },
+    },
+    watch: {
+      // eslint-disable-next-line no-unused-vars
+      partnersCount(val) {
+        console.log(`partnersCount(${val})`);
+        this.loadRooms();
+      },
+      // eslint-disable-next-line no-unused-vars
+      messagesCount(val) {
+        console.log(`messagesCount(${val})`);
+        // reload the current room
+        this.fetchMessages({room: { roomId: this.roomId}})
+      },
+    },
+
+    methods: {
+      ...mapMutations([
+        'message'
+      ]),
+      loadRooms() {
+        console.log("loadRooms()");
+        this.roomsLoaded = false;
+        this.rooms = [];
+
+        const _rooms = [];
+        const partners = this.$store.getters.getPartners;
+        for (let i = 0; i < partners.length; i++) {
+          const p = partners[i];
+          // assume they have a connection id, but check to make sure this partner is ARIES
+          if (p.ariesSupport) {
+            const name = partnerUtils.getPartnerName(p);
+            // each room is for a single partner/connection
+            // so set the room id to the partner id.
+            // add users to represent the partner and us.
+            const room = {
+              roomId: p.id,
+              roomName: name,
+              users: [
+                {
+                  _id: p.id,
+                  username: name
+                },
+                {
+                  _id: CHAT_CURRENT_USERID,
+                  username: "Me"
+                }
+              ]
+            };
+            _rooms.push(room);
+          }
+        }
+        _rooms.sort((a,b) => a.roomName.localeCompare(b.roomName))
+        this.rooms = _rooms;
+        this.roomId = this.roomId === null ? (_rooms.length ? _rooms[0].id : null) : null;
+        this.roomsLoaded = true;
+      },
+      // eslint-disable-next-line no-unused-vars
+      fetchMessages({ room, options = {} }) {
+        console.log(`fetchMessages(room = ${room.roomId})`);
+        this.messagesLoaded = false;
+        this.messages = [];
+        this.roomId = room.roomId;
+
+        const _msgs = [];
+        const allMessages = this.$store.getters.getPartnerMessages.map((x) => x);
```

Comment:
does that map((x)=>x) do anything? Looks to me like an identity function. I am not used to javascript.

Reply:
yes, this is just a way of doing a deep copy of the array.